docs/Plugin-Support.md

```diff
@@ -0,0 +1,583 @@
+# Plugin Support
+
+> :construction: The plugin support is **not stable or fully implemented as long as this warning is here**.
+
+VersionPress needs to understand plugins' data, actions, shortcodes and other things to automatically provide version control for them. If you're a plugin developer or enthusiast, this document is for you.
+
+> Note that *themes* are technically similar and will be supported in pretty much the same way, however, we focus on plugins first.
+
+
+## Introduction
+
+Plugins are described to VersionPress by a set of files stored in the `.versionpress` folder in the plugin root (other discovery options are described below). They include:
+
+- Actions in `actions.yml`
+- Database schema in `schema.yml`
+- Shortcodes in `shortcodes.yml`
+- Hooks are defined in `hooks.php`
+
+All files are optional so for example, if a plugin doesn't define any new shortcodes it can omit the `shortcodes.yml` file. Simple plugins like _Hello Dolly_ might even omit everything; they just need to have the `.versionpress` folder so that VersionPress knows the plugin is supported.
+
```

Comment:
I don't really know. Originally, I didn't intend to use this directory to let VP know if the plugin is supported. It's maybe question to @borekb.